src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -0,0 +1,29 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.service.CustomProductService;
+import com.es.phoneshop.service.ProductService;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+public class ProductDetailsPageServlet extends HttpServlet {
+    private ProductService productService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        productService = CustomProductService.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        long productId = Long.valueOf(request.getPathInfo().substring(1));
```

Comment:
what can return request.getPathInfo()?

Reply:
if there is no extra path information passed, this function will return null. Or it can be extra situation where it hasn't any numbers that we trying to get. I will handle this situation